Sources/FrontEnd/Parse/Parser.swift

```diff
@@ -2993,16 +2993,38 @@ public enum Parser {
     return head != nil ? try parseCompilerConditionTail(head: head!, in: &state) : nil
   }
 
+  /// Builds the sequence of compiler conditions for the present CompilerConditionTail
+  private static func buildSequenceCondition(
+    in state: inout ParserState
+  ) throws -> ConditionalCompilationStmt.SequenceCondition {
+    let condition = try parseCompilerCondition(in: &state)
+    let backup = state.backup()
+
+    // Look for the next `&&` or `||` operator if exists.
+    guard let firstOperHalf = state.take(), ["||", "&&"].contains(state.token(firstOperHalf).value)
+    else {
+      state.restore(from: backup)
+      return .operand(condition)
+    }
+
+    if state.token(firstOperHalf).value == "&&" {
+      return .and(condition, try buildSequenceCondition(in: &state))
+    } else {
+      return .or(condition, try buildSequenceCondition(in: &state))
+    }
+  }
+
   /// Parses a compiler condition structure, after the initial token (#if or #elseif).
   private static func parseCompilerConditionTail(
     head: Token, in state: inout ParserState
   ) throws -> AnyStmtID {
     // Parse the condition.
-    let condition = try parseCompilerCondition(in: &state)
+
+    let conditionSeq = try buildSequenceCondition(in: &state)
```

Comment:
Please avoid using abbreviations and suffixing variables by their type.
```suggestion
    let conditions = try buildSequenceCondition(in: &state)
```